webpack/assets/javascripts/react_app/components/ForemanModal/__tests__/__snapshots__/ForemanModal.integration.test.js.snap

```diff
@@ -0,0 +1,107 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`ForemanModal - integration tests should add, open, and close modals as directed by Redux actions: after closing modal1 1`] = `
+Object {
+  "action": Array [
+    Array [
+      Object {
+        "payload": Object {
+          "id": "modal1",
+        },
+        "type": "SET_MODAL_CLOSED",
+      },
+    ],
+  ],
+  "state": Object {
+    "foremanModals": Object {
+      "modal1": Object {
+        "open": false,
+      },
+      "modal2": Object {
+        "open": true,
+      },
+      "modal3": Object {
+        "open": false,
+      },
+    },
+  },
+}
+`;
+
+exports[`ForemanModal - integration tests should add, open, and close modals as directed by Redux actions: after opening modal1 1`] = `
+Object {
+  "action": Array [
+    Array [
+      Object {
+        "payload": Object {
+          "id": "modal1",
+        },
+        "type": "SET_MODAL_OPEN",
+      },
+    ],
+  ],
+  "state": Object {
+    "foremanModals": Object {
+      "modal1": Object {
+        "open": true,
+      },
+      "modal2": Object {
+        "open": false,
+      },
+      "modal3": Object {
+        "open": false,
+      },
+    },
+  },
+}
+`;
+
+exports[`ForemanModal - integration tests should add, open, and close modals as directed by Redux actions: after opening modal2 1`] = `
+Object {
+  "action": Array [
+    Array [
+      Object {
+        "payload": Object {
+          "id": "modal2",
+        },
+        "type": "SET_MODAL_OPEN",
+      },
+    ],
+  ],
+  "state": Object {
+    "foremanModals": Object {
+      "modal1": Object {
+        "open": true,
+      },
+      "modal2": Object {
+        "open": true,
```

Comment:
Wondering if it make sense to have 2 modals open at the same time. What would be the result of this scenario in real life? Should we allow it?

Reply:
@sharvit It's already in use in https://github.com/Katello/katello/pull/8384 where you can have the ManageManifestModal and the delete confirmation modal open at the same time.

Reply:
Does it work because the css z-index tweaked to allow it in this scenario?
I assume it won't work if I will just open 2 or more modals one by one?

Reply:
@sharvit Looking at the elements created by the PF3 component, it seems that all PF3 modals are assigned the same z-index:
```css
.modal {
  z-index: 1050;
}
```
When this is the case, (and assuming they are in the same [stacking context](https://tiffanybbrown.com/2015/09/css-stacking-contexts-wtf/), which I believe they are) the elements will stack in the [same order that they appear in the DOM](https://css-tricks.com/almanac/properties/z/z-index/).  So in this case, since the delete confirmation appears in the markup after the ManageManifestModal, it will be on top.

For this use case (having a main modal on screen and also a confirmation dialog for an action on the main modal) this behavior makes it easy for us, because we don't have to do any manual changes.

For an incorrect use case (trying to have two main modals on the screen) the consumer may see mixed results for which modal ends up on top.  However, since we want to discourage this type of use anyway, I don't think any changes are necessary.